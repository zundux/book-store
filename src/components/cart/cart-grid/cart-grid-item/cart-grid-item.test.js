import React from "react";
import CartGridItem, { AmountSelector, AuthorsList, AuthorsListItem, Price } from "./cart-grid-item";

describe("Cart grid item", () => {

  test("Cart grid item", () => {
    const product = {
      volumeInfo: {
        title: "title",
        imageLinks: {
          smallThumbnail: "img-src"
        },

        authors: ["author 1", "author 2"]
      },
      saleInfo: {
        listPrice: {
          amount: 1,
          currencyCode: "USD"
        },
        country: "EN"
      }
    };
    const onRemoveFromCart = jest.fn();
    const onChangeAmountOfProductInCart = () => {
    };

    const component = shallow(
      <CartGridItem {...{
        product,
        onRemoveFromCart,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);

    component.find(".remove-product").simulate("click");
    expect(onRemoveFromCart).toHaveBeenCalled();
    expect(onRemoveFromCart).toBeCalledWith(product);
  });


  test("Product amount selector", () => {
    const product = {
      amount: 1
    };
    const onChangeAmountOfProductInCart = jest.fn();

    const component = shallow(
      <AmountSelector {...{
        product,
        onChangeAmountOfProductInCart
      }}/>
    );
    matchSnapshot(component);

    component.find(".decrease").simulate("click");
    expect(onChangeAmountOfProductInCart).toHaveBeenCalled();
    expect(onChangeAmountOfProductInCart).toBeCalledWith(product, product.amount - 1);

    component.find(".increase").simulate("click");
    expect(onChangeAmountOfProductInCart.mock.calls.length).toEqual(2);
    expect(onChangeAmountOfProductInCart).toBeCalledWith(product, product.amount + 1);

    const amountField = component.find("[name=\"amount\"]");
    expect(amountField.props().value).toBe(1);
    const exactAmount = 13;
    const changeEvent = { target: { name: "amount", value: exactAmount } };
    amountField.simulate("change", changeEvent);
    expect(onChangeAmountOfProductInCart.mock.calls.length).toEqual(3);
    expect(onChangeAmountOfProductInCart).toBeCalledWith(product, exactAmount);
  });


  test("Product price EUR", () => {
    const saleInfo = {
      listPrice: {
        amount: 1,
        currencyCode: "EUR"
      },
      country: "FR"
    };

    const component = shallow(
      <Price {...{
        saleInfo
      }}/>
    );
    matchSnapshot(component);
  });

  test("Product price USD", () => {
    const saleInfo = {
      listPrice: {
        amount: 2,
        currencyCode: "USD"
      },
      country: "EN"
    };

    const component = shallow(
      <Price {...{
        saleInfo
      }}/>
    );
    matchSnapshot(component);
  });


  test("Authors list - no authors", () => {
    const authors = [];

    const component = shallow(
      <AuthorsList {...{
        authors
      }}/>
    );
    matchSnapshot(component);
  });

  test("Authors list - one author", () => {
    const authors = ["author 1"];

    const component = shallow(
      <AuthorsList {...{
        authors
      }}/>
    );
    matchSnapshot(component);
  });

  test("Authors list - a couple of authors", () => {
    const authors = ["author 1", "author 2"];

    const component = shallow(
      <AuthorsList {...{
        authors
      }}/>
    );
    matchSnapshot(component);
  });

  test("Authors list - a lot of authors", () => {
    const authors = [
      "author 1", "author 2", "author 3", "author 4",
      "author 5", "author 6", "author 7", "author 8",
      "author 9", "author 10", "author 11", "author 12",
      "author 13", "author 14", "author 15", "author 16",
      "author 17", "author 18", "author 19", "author 20"
    ];

    const component = shallow(
      <AuthorsList {...{
        authors
      }}/>
    );
    matchSnapshot(component);
  });


  test("Authors list item", () => {
    const author = "author 1";
    const authorIndex = 1;

    const component = shallow(
      <AuthorsListItem {...{
        author,
        authorIndex
      }}/>
    );
    matchSnapshot(component);
  });

});