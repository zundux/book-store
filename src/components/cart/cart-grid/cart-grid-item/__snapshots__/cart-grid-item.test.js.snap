// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart grid item Authors list - a couple of authors 1`] = `
<div>
  <p>
    Authors
    :
  </p>
  <ul>
    <li
      key="0"
    >
      author 1
    </li>
    <li
      key="1"
    >
      author 2
    </li>
  </ul>
</div>
`;

exports[`Cart grid item Authors list - a lot of authors 1`] = `
<div>
  <p>
    Authors
    :
  </p>
  <ul>
    <li
      key="0"
    >
      author 1
    </li>
    <li
      key="1"
    >
      author 2
    </li>
    <li
      key="2"
    >
      author 3
    </li>
    <li
      key="3"
    >
      author 4
    </li>
    <li
      key="4"
    >
      author 5
    </li>
    <li
      key="5"
    >
      author 6
    </li>
    <li
      key="6"
    >
      author 7
    </li>
    <li
      key="7"
    >
      author 8
    </li>
    <li
      key="8"
    >
      author 9
    </li>
    <li
      key="9"
    >
      author 10
    </li>
    <li
      key="10"
    >
      author 11
    </li>
    <li
      key="11"
    >
      author 12
    </li>
    <li
      key="12"
    >
      author 13
    </li>
    <li
      key="13"
    >
      author 14
    </li>
    <li
      key="14"
    >
      author 15
    </li>
    <li
      key="15"
    >
      author 16
    </li>
    <li
      key="16"
    >
      author 17
    </li>
    <li
      key="17"
    >
      author 18
    </li>
    <li
      key="18"
    >
      author 19
    </li>
    <li
      key="19"
    >
      author 20
    </li>
  </ul>
</div>
`;

exports[`Cart grid item Authors list - no authors 1`] = `
<div>
  <p>
    Author
    :
  </p>
  <p>
    no information
  </p>
</div>
`;

exports[`Cart grid item Authors list - one author 1`] = `
<div>
  <p>
    Author
    :
  </p>
  <ul>
    <li
      key="0"
    >
      author 1
    </li>
  </ul>
</div>
`;

exports[`Cart grid item Authors list item 1`] = `
<li
  key="[object Object]"
>
  <Component />
</li>
`;

exports[`Cart grid item Cart grid item 1`] = `
<div>
  <img
    alt="title"
    src="img-src"
  />
  <p>
    <strong>
      title
    </strong>
  </p>
  <Component
    authors={
      Array [
        "author 1",
        "author 2",
      ]
    }
  />
  <Component
    saleInfo={
      Object {
        "country": "EN",
        "listPrice": Object {
          "amount": 1,
          "currencyCode": "USD",
        },
      }
    }
  />
  <Component
    onChangeAmountOfProductInCart={[Function]}
    product={
      Object {
        "saleInfo": Object {
          "country": "EN",
          "listPrice": Object {
            "amount": 1,
            "currencyCode": "USD",
          },
        },
        "volumeInfo": Object {
          "authors": Array [
            "author 1",
            "author 2",
          ],
          "imageLinks": Object {
            "smallThumbnail": "img-src",
          },
          "title": "title",
        },
      }
    }
  />
  <button
    className="remove-product"
    onClick={[Function]}
  >
    Remove from cart
  </button>
</div>
`;

exports[`Cart grid item Product amount selector 1`] = `
<div>
  <button
    className="decrease"
    onClick={[Function]}
  >
    -
  </button>
  <input
    name="amount"
    onChange={[Function]}
    type="number"
    value={1}
  />
  <button
    className="increase"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`Cart grid item Product price EUR 1`] = `
<div>
  <span>
    â‚¬1.00
  </span>
</div>
`;

exports[`Cart grid item Product price USD 1`] = `
<div>
  <span>
    $2.00
  </span>
</div>
`;
